File: UaicLibrary.Domain/BookManagement/BookFilter.cs
using System.Collections.Generic;
using UaicLibrary.Common.Filters;
using UaicLibrary.Domain.GeneralModels;

namespace UaicLibrary.Domain.BookManagement
{
    public class BookFilter : PaginationFilter
    {
        public string Title { get; set; }
        public IList<SelectItemModel> Categories { get; set; }
        public IList<SelectItemModel> Authors { get; set; }
        public IList<SelectItemModel> Faculties { get; set; }
    }
}

File: UaicLibraryWebSpa/src/app/filters/book.filter.ts
import { PaginationFilter } from './pagination.filter';
import { MultiSelectItem } from '../components/common/multiselect/multi-select-item';

export class BookFilter extends PaginationFilter {
    constructor(public page: number, public pageSize: number, public categories: Array<MultiSelectItem>,
        public authors: Array<MultiSelectItem>, public faculties: Array<MultiSelectItem>, public title: string) {
        super(page, pageSize);
    }
}